src/Umbraco.Tests/Testing/ContentBaseExtensions.cs

```diff
@@ -1,25 +1,29 @@
 ﻿using System;
 using System.Linq;
+using Umbraco.Core;
+using Umbraco.Core.Composing;
 using Umbraco.Core.Models;
 
 namespace Umbraco.Tests.Testing
 {
     public static class ContentBaseExtensions
     {
         /// <summary>
-        /// Set property values by alias with an annonymous object.
+        /// Set property values by alias with an anonymous object.
         /// </summary>
         /// <remarks>Does not support variants.</remarks>
         public static void PropertyValues(this IContentBase content, object value, string culture = null, string segment = null)
         {
             if (value == null)
                 throw new Exception("No properties has been passed in");
 
+            var contentType = Current.Services.ContentTypeBaseServices.GetContentTypeOf(content);
+
             var propertyInfos = value.GetType().GetProperties();
             foreach (var propertyInfo in propertyInfos)
             {
                 //Check if a PropertyType with alias exists thus being a valid property
-                var propertyType = content.PropertyTypes.FirstOrDefault(x => x.Alias == propertyInfo.Name);
+                var propertyType = contentType.CompositionPropertyTypes.FirstOrDefault(x => x.Alias == propertyInfo.Name);
```

Comment:
Seems like we don't need to do this check either, we will already know if the property exist by that alias in the `content.Properties` collection which means we don't have to look up the content type unless we are creating a new property below.